test/Dialect/Comb/canonicalization.mlir

```diff
@@ -1533,3 +1533,22 @@ hw.module @OrMuxSameTrueValueAndZero(in %tag_0: i1, in %tag_1: i1, in %tag_2: i1
   %4 = comb.or bin %0, %1, %2, %3 : i4
   hw.output %4 : i4
 }
+
+// CHECK-LABEL:   "test.acrossBlockCanonicalizationBarrier"() ({
+// CHECK:         ^bb0(%[[A:.*]]: i32, %[[B:.*]]: i32, %[[C:.*]]: i32):
+// CHECK:           %[[ADD1:.*]] = comb.add %[[A]], %[[B]] : i32
+// CHECK:           "terminator"() : () -> ()
+// CHECK:         ^bb1:
+// CHECK:           %[[ADD2:.*]] = comb.add %[[ADD1]], %[[C]] : i32
+// CHECK:           "terminator"(%[[ADD2]]) : (i32) -> ()
+// CHECK:         }) : () -> ()
+"test.acrossBlockCanonicalizationBarrier"() ({
```

Comment:
I might like to see more tests. OTOH given the approach, might not be necessary.